src/server/routes/query/query.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2017-2022 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Request, Response, Router } from "express";
+import { Dataset } from "plywood";
+import { deserialize } from "../../../client/deserializers/app-settings";
+import makeGridQuery from "../../../client/visualizations/grid/make-query";
+import { errorToMessage } from "../../../common/logger/logger";
+import { AppSettings, ClientAppSettings, serialize } from "../../../common/models/app-settings/app-settings";
+import { ClientDataCube } from "../../../common/models/data-cube/data-cube";
+import { isQueryable, QueryableDataCube } from "../../../common/models/data-cube/queryable-data-cube";
+import { Essence } from "../../../common/models/essence/essence";
+import { getDataCube, Sources } from "../../../common/models/sources/sources";
+import makeQuery from "../../../common/utils/query/visualization-query";
+import { definitionConverters, ViewDefinitionVersion } from "../../../common/view-definitions";
+import { checkAccess } from "../../utils/datacube-guard/datacube-guard";
+import { loadQueryDecorator } from "../../utils/query-decorator-loader/load-query-decorator";
+import { SettingsManager } from "../../utils/settings-manager/settings-manager";
+
+function convertToClientAppSettings(appSettings: AppSettings): ClientAppSettings {
+  return deserialize(serialize(appSettings));
+}
+
+function convertToClientDataCube(cube: QueryableDataCube): ClientDataCube {
+  return {
+    ...cube,
+    timeAttribute: cube.timeAttribute && cube.timeAttribute.name
+  };
+}
+
+export function queryRouter(settings: Pick<SettingsManager, "logger" | "getSources" | "appSettings" | "anchorPath" | "getTimekeeper">) {
```

Comment:
A lot of duplication from /plywood and /mkurl endpoints. Maybe it's time to have good validation helpers?